python/ray/dag/compiled_dag_node.py

```diff
@@ -438,30 +455,50 @@ def _compute(self, class_handle) -> bool:
             # Propagate it and skip the actual task. We don't need to wrap the
             # exception in a RayTaskError here because it has already been wrapped
             # by the previous task.
-            self.set_intermediate_buffer(exc)
+            self.set_intermediate_buffer(operation, exc)
             return False
 
+        channel_results = []
+        for entry in input_data:
+            if isinstance(entry, GPUFuture):
+                channel_result = entry.wait()
+            else:
+                channel_result = entry
+            channel_results.append(channel_result)
+
         resolved_inputs = []
         for task_input in self.task_inputs:
-            resolved_inputs.append(task_input.resolve(input_data))
+            resolved_inputs.append(task_input.resolve(channel_results))
```

Comment:
Perhaps we should be calling `.wait()` in `task_input.resolve` instead of in this function?

Reply:
Done